openai/api_requestor.py

```diff
@@ -149,6 +151,64 @@ def format_app_info(cls, info):
             str += " (%s)" % (info["url"],)
         return str
 
+    def _check_polling_response(self, response: OpenAIResponse, predicate: Callable[[OpenAIResponse], bool]):
+        if not predicate(response):
+            return
+        error_data = response.data['error']
+        message = error_data.get('message', 'Operation failed')
+        code = error_data.get('code')
+        raise error.OpenAIError(message=message, code=code)
+
+    def _poll(
+        self,
+        method,
+        url,
+        until,
+        failed,
+        params = None,
+        headers = None,
+        interval = None,
+        delay = None
+    ) -> Tuple[Iterator[OpenAIResponse], bool, str]:
+        if delay:
+            time.sleep(delay)
+
+        response, b, api_key = self.request(method, url, params, headers)
+        self._check_polling_response(response, failed)
+        start_time = time.time()
+        while not until(response) and time.time() - start_time < TIMEOUT_SECS:
```

Comment:
I'm pushing back a bit but these are mostly nits, feel free to steamroll them if you have stronger guarantees about the API response than I know about. But, if the timeout fails and we break the loop, the following line to index into `response.data['result']` would raise a `KeyError`.

Reply:
update it to raise an exception.